galleries/examples/event_handling/viewlims.py

```diff
@@ -59,15 +52,35 @@ def ax_update(self, ax):
         self.width, self.height = \
             np.round(ax.patch.get_window_extent().size).astype(int)
         # Get the range for the new area
-        vl = ax.viewLim
-        extent = vl.x0, vl.x1, vl.y0, vl.y1
+        x0, x1 = ax.get_xlim()
+        y0, y1 = ax.get_ylim()
+        extent = x0, x1, y0, y1
         # Update the image object with our new data and extent
         im = ax.images[-1]
         im.set_data(self.compute_image(*extent))
         im.set_extent(extent)
         ax.figure.canvas.draw_idle()
 
 
+def select_callback(eclick, erelease):
+    extent = rect_selector.extents
+
+    ax2.set_autoscale_on(False)
+    # Zoom the selected part
+    # Set xlim, ylim range for plot
+    # of rectangle selector box.
+    ax2.set_xlim(extent[0], extent[1])
+    ax2.set_ylim(extent[2], extent[3])
+
+    # update the right subplot
+    md = MandelbrotDisplay()
```

Comment:
You should not make a new object, but update on the existing one, as was done before.